### PR TITLE
Increase coverage for tracers

### PR DESCRIPTION
**Pull request by MakisChristou**
# Description
Increased coverage of the following files:
`gen_structlog.go`: 36.96% -> 71.7%
`logger.go`: 26.06% -> 70.4%
`logger_json.go`: 0.00% -> 78.9%

Fixes [#21](https://github.com/vechain/protocol-board-repo/issues/21)

### CODE REVIEW

File: tracers/logger/gen_structlog_test.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package logger
```
Comment (libotony) [inline tracers/logger/gen_structlog_test.go:1]: `gen_structlog.go` is generated file, tests should be ignored

File: tracers/logger/logger_test.go [hunk 2]
```diff
@@ -105,3 +116,111 @@ func TestStructLogMarshalingOmitEmpty(t *testing.T) {
...
+	logger.CaptureClauseEnd(1234)
+}
+
+func TestWriteLogs(t *testing.T) {
+	writer := &bytes.Buffer{}
+
+	logs := []*types.Log{
+		{
+			Address:     common.HexToAddress("0x1"),
+			Topics:      []common.Hash{mockHash("topic1"), mockHash("topic2")},
```
Comment (libotony) [inline tracers/logger/logger_test.go:168]: `common.BytesToHash([]byte("topic1"))`

Comment (MakisChristou) [reply tracers/logger/logger_test.go:168]: Can you elaborate a bit on this one?

Comment (libotony) [reply tracers/logger/logger_test.go:168]: The function `mockHash` can be replaced by common.BytesToHash

Comment (MakisChristou) [reply tracers/logger/logger_test.go]: Let me know if this is what you wanted

**Review by libotony — APPROVED** [review]

**Pull request merged**